File: pre-contest/b_the_alchemist/sol/sol.cs
using System;

class b_the_alchemist {
    public static void Main() 
    {
        var input = Console.ReadLine();
        var inputParts = input.Split(' ');
        int p = int.Parse(inputParts[0]);
        int l = int.Parse(inputParts[1]);
        int g = int.Parse(inputParts[2]);
        
        int cpt = 0;
        for(int x = 0; x <= p && x*x <= g; x++) {
            for(int y=0; y < l; y++){
                int r = x*x + x*y + y;
                if(r == g) {
                    cpt++;
                }
                else if(r > g) {
                    break;
                }
            }
        }
        
        Console.Write(cpt);
    }
}

File: pre-contest/b_the_alchemist/sol/sol.rs
use std::io;
use std::f32;
use std::cmp;

fn main() {
    let mut line = String::new();
    io::stdin().read_line(&mut line).unwrap();

    let v: Vec<i32> = line.trim()
        .split(' ')
        .map(|x| x.parse::<i32>().unwrap())
        .collect();
    let (p, l, g) = (v[0], v[1], v[2]);

    let max_x = cmp::min((g as f32).sqrt().ceil() as i32, p);

    let mut counter = 0;
    for x in 0..max_x {
        for y in 0..l {
            let r = x * x + y + x * y;
            if r == g {
                counter += 1;
            } else if r > g {
                break;
            }
        }
    }

    println!("{}", counter);
}

